### ISSUE TITLE
testCMCWithControlConstraintsRunningModel periodically fails on appveyor

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#350]
The first time I started seeing this was with the recent Frames PR. I'm not saying the Frames PR caused it, though.

**Comment by chrisdembia** [comment]
This is the message that I see on appveyor:

```
61/63 Test #62: testCMCWithControlConstraintsRunningModel ....***Exception: SegFault 63.70 sec
 
******************************************************************
*                     testRunningModel                     *
******************************************************************
 
AbstractTool runningModel_CMC_test loading model 'runningModel_CMC_test.osim'
Updating Model file from 20302 to latest format...
Loaded model modelModified from file runningModel_CMC_test.osim
Adding force object set from runningModel_CMC_Actuators.xml
Running tool runningModel_CMC_test.
Old version storage/motion file encountered
Storage: file=runningModel_GRF_data.mot (nr=691 nc=19)
.. assuming rotations in Degrees.
ExternalForce::RightFoot_ExternalForce
Data source being set to runningModel_GRF_data.mot
ExternalForce::LeftFoot_ExternalForce
Data source being set to runningModel_GRF_data.mot
```

**Comment by chrisdembia** [comment]
Can someone on a windows machine run this test a few times to see if it fails periodically?

**Comment by tkuchida** [comment]
Passed 4 consecutive times.

**Comment by tkuchida** [comment]
The next line in the output is `taskSet size = 29`… maybe the task set file isn't being read properly? No hypothesis as to why that would cause a periodic failure, though.

**Comment by chrisdembia** [comment]
Thanks for looking into this. Yeah...I don't know. And every time I've rerun the appveyor build after failure, it succeeds.

Maybe appveyor is just timing out and I didn't notice?

**Comment by chrisdembia** [comment]
Now testCMCGait10dof18musc failed: https://ci.appveyor.com/project/opensim-org/opensim-core/build/192

```
AbstractTool walk_subject loading model 'gait10dof18musc_subject01.osim'
Updating Model file from 30000 to latest format...
Loaded model walk_subject01 from file gait10dof18musc_subject01.osim
Adding force object set from gait10dof_Reserve_Actuators.xml
Running tool walk_subject.
Storage: file=gait10dof18musc_subject01_walk_grf.mot (nr=1501 nc=19)
ExternalForce::right
Data source being set to subject01_walk1_grf.mot
ExternalForce::left
Data source being set to subject01_walk1_grf.mot
```

Seems to fail in a similar way. I agree with Tom that it might be a file reading error.

**Comment by chrisdembia** [comment]
@aseth1 mentioned it might be timing: two tests might be concurrently be writing to the same file.

**Comment by aymanhab** [comment]
@chrisdembia Writing the same file sounds scary, how do we ever know that the tests are really executing? Can the files go into separate subdirectories?

**Comment by chrisdembia** [comment]
> Can the files go into separate subdirectories?

Great idea! We can do something like this, yeah.

**Comment by chrisdembia** [comment]
My current idea for fixing this is to get OpenSim to print a stack backtrace when segfaulting. http://oroboro.com/stack-trace-on-crash/

**Comment by sherm1** [comment]
> print a stack backtrace when segfaulting

+1

**Comment by chrisdembia** [comment]
> +1

Do you think this would be a good idea in general, aside from fixing this particular issue? This would definitely help with users filing bug reports (btw I think our releases should be RelWithDebInfo), but I think it might be really scary to users. Maybe the stack backtrace should be written to yet another file (not to the console).

**Comment by sherm1** [comment]
> Do you think this would be a good idea in general [outputting stack trace on crash]

I think we should play with it first to see whether anything useful comes out. I don't know if it would actually help us debug user problems, but I agree with you that if we do it it should go into a separate log file. Then we can tell them to upload it if they report a crash.

**Comment by chrisdembia** [comment]
Here might be a hint from the opensim32 repo:

```
AbstractTool runningModel_CMC_test loading model 'runningModel_CMC_test.osim'
Updating Model file from 20302 to latest format...
Loaded model modelModified from file runningModel_CMC_test.osim
Adding force object set from runningModel_CMC_Actuators.xml
Running tool runningModel_CMC_test.
Old version storage/motion file encountered
Storage: file=runningModel_GRF_data.mot (nr=691 nc=19)
.. assuming rotations in Degrees.
ExternalForce::RightFoot_ExternalForce
Data source being set to runningModel_GRF_data.mot
ExternalForce::LeftFoot_ExternalForce
Data source being set to runningModel_GRF_data.mot
 
52/52 Test #50: testCMC ......................................   Passed  379.26 sec
 
98% tests passed, 1 tests failed out of 52
 
Total Test time (real) = 530.06 sec
 
The following tests FAILED:
	 51 - testCMCWithControlConstraints (SEGFAULT)
Errors while running CTest
Command exited with code 8
```

This might have to do with `Old version storage/motion file encountered`

**Comment by tkuchida** [comment]
[testCMCGait10dof18musc failed on AppVeyor x64](https://ci.appveyor.com/project/opensim-org/opensim-core/build/3986/job/8kky5dh3oyku2uhg#L999) in #1072:
```
99% tests passed, 1 tests failed out of 78
Total Test time (real) = 247.58 sec
The following tests FAILED:
	 66 - testCMCGait10dof18musc (SEGFAULT)
Errors while running CTest
Command exited with code 8
```

**Comment by chrisdembia** [comment]
This might be addressed by #1333.
